src/pages/LessonPage.tsx

```diff
@@ -1,59 +1,107 @@
-    borderColor: '#04364A',
-    color: '#DAFFFB',
-  },
-  '&:active': {
-    backgroundColor: '#04364A',
-    borderColor: '#04364A',
-  },
-})
-
-export default function LessonPage(props: LessonPageProps) {
-  const [currentBlock, setCurrentBlock] = useState<number>(0)
+function getWindowDimensions() {
+  const { innerWidth: width, innerHeight: height } = window
+  return {
+    width,
+    height,
+  }
+}
+
+function SideNavigatorItem({ selected, onClick }) {
+  return (
+    <div
+      onClick={onClick}
+      className="fadeColor"
+      style={{
+        marginTop: 10,
+        height: selected ? 100 : 40,
+        backgroundColor: selected ? '#04364a' : '#6DB6C3',
+        borderRadius: 50,
+        cursor: 'pointer',
+      }}
+    />
+  )
+}
+
+export default function LessonPage({ blocks }: LessonPageProps) {
+  const refs = useRef(new Array(blocks.length))
+  const [offsetY, setOffsetY] = useState(0)
+  const [windowDimensions, setWindowDimensions] = useState(
+    getWindowDimensions(),
+  )
+
+  const currentIndex = () => Math.round(offsetY / windowDimensions.height)
+
+  useEffect(() => {
+    const handleScroll = () => {
+      setOffsetY(window.scrollY)
+    }
+    window.addEventListener('scroll', handleScroll)
+    return () => window.removeEventListener('scroll', handleScroll)
+  }, [])
+
+  useEffect(() => {
+    const handleResize = () => setWindowDimensions(getWindowDimensions())
+    window.addEventListener('resize', handleResize)
+    return () => window.removeEventListener('resize', handleResize)
+  }, [])
 
   return (
-    <Container>
-      {props.blocks[currentBlock]}
-      <Box sx={{ display: 'flex', justifyContent: 'space-between' }}>
-        <LessonButton
-          onClick={() =>
-            setCurrentBlock((blockIndex) =>
-              blockIndex - 1 >= 0 ? blockIndex - 1 : 0,
-            )
-          }
-        >
-          Back
-        </LessonButton>
-        <LessonButton
-          onClick={() =>
-            setCurrentBlock(
-              (blockIndex) => (blockIndex + 1) % props.blocks.length,
-            )
-          }
-        >
-          Next
-        </LessonButton>
+    <Box sx={{ position: 'relative' }}>
+      <Box
+        sx={{
+          height: 'calc(100vh)',
+        }}
+      >
+        {blocks.map((block, id) => {
+          return (
+            <Container
+              key={id}
+              sx={{
+                scrollSnapAlign: 'start',
+                height: '100%',
+                overflowY: 'auto',
+                pt: 15,
+              }}
+              ref={(element) => {
+                refs.current[id] = element
+              }}
+            >
+              {block}
+            </Container>
+          )
+        })}
+      </Box>
+
+      <Box
+        sx={{
+          width: 20,
+          transform: 'translateY(-50%)',
+          position: 'fixed',
+          top: '50%',
+          left: '20px',
+        }}
+      >
+        {blocks.map((_item, id) => {
```

Comment:
just nit: do we need to have separate scroll bar for block if it has extra content or should whole lesson page have single scroll bar to right, like for the DP intro and Example Code Reading Block there is another scroll bar just for those blocks
<img width="1552" alt="Screenshot 2023-11-07 at 9 53 42 PM" src="https://github.com/ofast-team/ofast/assets/84353905/deb4bbc5-7e06-43f8-a7a4-8152b7a124ea">

Reply:
This is because I set the lesson block heights to a constant 100vh. I was going to fix this in another diff since it's tricky to have varying heights depending on the content, but I just figured it out and pushed a fix.